emloop/hooks/benchmarking.py

```diff
@@ -0,0 +1,28 @@
+"""
+Hook for benchmarking models.
+"""
+import logging
+import numpy as np
+from typing import List
+
+from . import AbstractHook
+from ..types import TimeProfile
+
+
+class Benchmarking(AbstractHook):
+    def __init__(self, batch_size, dataset, model, output_dir, **kwargs):
+        super().__init__(**kwargs)
+        self._batch_size = batch_size
+        self._dataset = dataset
+        self._model = model
+        self._output_dir = output_dir
+
+    def after_epoch_profile(self, epoch_id: int, profile: TimeProfile, streams: List[str]):
+        """Prints average time per example."""
+        for stream_name in streams:
+            batch_times = profile.get(f'eval_batch_{stream_name}')
+            # last batch may be smaller than the other ones, so we drop it to not skew the measurement
+            del batch_times[-1]
+            example_times = list(map(lambda x: x / float(self._batch_size), batch_times))
```

Comment:
can't we just use `batch_times[:-1]` and remove `del batch_times[-1]` line? I really don't like using `del`